### PR TITLE
ddl : ddl forbid create / add / rename /drop index for cache table 

### LINKED ISSUE #29659: For cache table should not be allowed to create indexes
This ia subtask  #25293
for cache table .` create index xxx on(cache_table)` should not be allowed.
so tne following operator is valid.
```
alter table cache_table nocache
create index xxx on (cache_table)
alter table cache_table cache
```

### PR DESCRIPTION
**Pull request by JayL-zxl**
Issue Number: close #29659
Problem Summary:
This is a subtask of https://github.com/pingcap/tidb/issues/25293

### What is changed and how it works?
+ Index operator shoud forbid for cache table
+ The following operator is valid.
```
Alter table t nocache
.... do some index operator
```
### Check List 

Tests 

-  Unit test

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: ddl/index.go [hunk 1]
```diff
@@ -304,6 +304,9 @@ func onRenameIndex(t *meta.Meta, job *model.Job) (ver int64, _ error) {
 	if err != nil || tblInfo == nil {
 		return ver, errors.Trace(err)
 	}
+	if tblInfo.TableCacheStatusType != model.TableCacheStatusDisable {
```
Comment (lcwangchao) [inline ddl/index.go:307]: Better to check it before schedule to avoid unnecessary job

Comment (JayL-zxl) [reply ddl/index.go:307]: you are right.

File: ddl/ddl_api.go [hunk 2]
```diff
@@ -5400,6 +5403,9 @@ func (d *ddl) CreateIndex(ctx sessionctx.Context, ti ast.Ident, keyType ast.Inde
 		return errors.Trace(err)
 	}
 
+	if t.Meta().TableCacheStatusType != model.TableCacheStatusDisable {
```
Comment (lcwangchao) [inline ddl/ddl_api.go:5406]: I think we should do checking both before and in schedule.

Comment (JayL-zxl) [reply ddl/ddl_api.go:5406]: you mean is double check?

**Review by lcwangchao — APPROVED** [review]

**Review by tiancaiamao — APPROVED** [review]

**Pull request merged**